docs/proposals/20220712-cluster-api-addon-orchestration.md

```diff
@@ -0,0 +1,445 @@
+  // +optional
+  Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+
+  // Status is the current status of the Helm release.
+  // +optional
+  Status string `json:"status,omitempty"`
+
+  // Revision is the current revision of the Helm release.
+  // +optional
+  Revision int `json:"revision,omitempty"`
+}
+```
+
+#### Controller Design
+
+The HelmChartProxy controller is responsible for maintaining our list of HelmReleaseProxies and resolving any referenced Cluster fields. The controller will:
+
+- Find all workload clusters matching the cluster selector within the same namespace.
+- Resolve the referenced Cluster fields in the values map based on each selected cluster.
+- Create a HelmReleaseProxy for each selected workload cluster if one does not exist with the resolved values.
+- Update the HelmReleaseProxy for each selected workload cluster if it exists and the version or resolved values have changed.
+- Delete the HelmReleaseProxy for each selected workload cluster if the cluster no longer matches the cluster selector.
+
+The HelmReleaseProxy controller is responsible for creating, updating, and deleting the Helm release installed on the workload cluster. The controller will:
+
+- Install a Helm release on the workload cluster if it does not exist.
+- Update the existing Helm release if the HelmReleaseProxy values or version do not match the existing Helm release.
+- Delete the Helm release if the HelmReleaseProxy is deleted.
+
+This design of HelmReleaseProxy is especially convenient for the controller because a `Create()` maps directly to a `helm install` operation, a `Update()` maps directly to a `helm upgrade` operation, and a `Delete()` maps directly to a `helm uninstall` operation.
+
+Note that the HelmReleaseProxy controller contains a static configuration and is only responsible for ensuring there is a Helm release matching the HelmReleaseProxy. The HelmReleaseProxy gets created, updated, and deleted by the HelmChartProxy controller.
+
+#### Defining the list of Cluster add-ons to install
+
+HelmChartProxy allows users to define the add-ons to install and which clusters they should be installed on using the `clusterSelector`. The specified Helm chart is then installed on every cluster that includes a label matching the cluster selector and makes it part of the orchestration. If a selected cluster no longer matches the cluster selector label, then the Helm release will be uninstalled. 
+
+#### Installing and lifecycle managing the package manager
+
+ClusterAddonProvider for Helm imports Helm v3 as a library, making this a no-op. However, it is worth noting that each version of Helm only supports certain version of Kubernetes. For example, the [Helm docs](https://helm.sh/docs/topics/version_skew/#supported-version-skew) mention that Helm 3.9 supports Kubernetes 1.21 through 1.24. As a result, there may need to be multiple versions of HelmChartProxy and HelmReleaseProxy in order to be backwards compatible.
```

Comment:
```suggestion
ClusterAddonProvider for Helm imports Helm v3 as a library, making this a no-op. However, it is worth noting that each version of Helm only supports certain version of Kubernetes. For example, the [Helm docs](https://helm.sh/docs/topics/version_skew/#supported-version-skew) mention that Helm 3.9 supports Kubernetes 1.21 through 1.24.
```

Sorry, I meant dropping this sentence as well in my comment.

I think it would be confusing to have multiple different apiVersions of HelmChartProxy and HelmReleaseProxy to cover different Kubernetes versions. The sentence afterwards was fine

>  In this case, we will document the version support matrix for ClusterAddonProvider for Helm if it differs from the CAPI support matrix.